[Bug]: AMM datanode panics on system-tests and market-sim fuzz

### Problem encountered

The latest change to allow low-volume AMM's has caused panics during the data-node's caculations of market-depths:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x320c472]
goroutine 1301121 [running]:
code.vegaprotocol.io/vega/core/execution/amm.(*Sqrter).sqrt(0xc003bf2580?, 0xc0ffffffe0?)
	/jenkins/workspace/common/system-tests-wrapper/vega/core/execution/amm/engine.go:99 +0x12
code.vegaprotocol.io/vega/core/execution/amm.EstimateBounds(0xc006274ab0, 0xc005c47540, 0xc005c474c0, _, {_, _}, {_, _}, _, {0xc003e2bdc0, ...}, ...)
	/jenkins/workspace/common/system-tests-wrapper/vega/core/execution/amm/estimator.go:78 +0x405
code.vegaprotocol.io/vega/datanode/api.(*TradingDataServiceV2).EstimateAMMBounds(0xc000c98000, {0x67a6008, 0xc002e27170}, 0xc0013ea580)
	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/api/trading_data_v2.go:6011 +0xc6a
```

and

```
panic: runtime error: index out of range [0] with length 0

goroutine 315 [running]:
code.vegaprotocol.io/vega/datanode/service.(*MarketDepth).expandByLevels(_, {{0xc002630ec0, 0x40}, {0xc002630f00, 0x40}, {0xc002630f40, 0x40}, {0xc002630f80, 0x40}, {0xc002769ac0, ...}, ...}, ...)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/datanode/service/market_depth_amm.go:248 +0x9d0
code.vegaprotocol.io/vega/datanode/service.(*MarketDepth).ExpandAMM(_, {_, _}, {{0xc002630ec0, 0x40}, {0xc002630f00, 0x40}, {0xc002630f40, 0x40}, {0xc002630f80, ...}, ...}, ...)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/datanode/service/market_depth_amm.go:419 +0x1b9
code.vegaprotocol.io/vega/datanode/service.(*MarketDepth).refreshAMM(_, {{0xc002630ec0, 0x40}, {0xc002630f00, 0x40}, {0xc002630f40, 0x40}, {0xc002630f80, 0x40}, {0xc002769ac0, ...}, ...}, ...)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/datanode/service/market_depth_amm.go:470 +0x453
```

### Observed behaviour

Data node panics.

### Expected behaviour

Data node does not panic.

### Steps to reproduce

```bash
1. Submit `xyz` transaction
2. With this payload
3. Query `abc` API
4. Observe the error
```


### Software version

develop

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/core/execution/amm/estimator.go
+++ b/core/execution/amm/estimator.go
@@ -75,7 +75,7 @@ func EstimateBounds(
 			l:        l,
 			high:     basePrice,
 			low:      lowerPrice,
-			sqrtHigh: sqrter.sqrt(upperPrice),
+			sqrtHigh: sqrter.sqrt(basePrice),
 			isLower:  true,
 			pv:       r.PositionSizeAtLower,
 		}

--- a/core/execution/amm/estimator_test.go
+++ b/core/execution/amm/estimator_test.go
@@ -294,5 +294,4 @@ func TestEstimateBoundsOneSided(t *testing.T) {
 	assert.True(t, metrics.LiquidationPriceAtLower.IsZero())
 	assert.True(t, metrics.PositionSizeAtLower.IsZero())
 	assert.True(t, metrics.LossOnCommitmentAtLower.IsZero())
-
 }

--- a/core/execution/amm/estimator_test.go
+++ b/core/execution/amm/estimator_test.go
@@ -238,3 +238,61 @@ func TestEstimatePositionFactor(t *testing.T) {
 	assert.False(t, metrics.TooWideLower) // is valid as there are less than 10 empty price levels
 	assert.True(t, metrics.TooWideUpper)  // isn't valid as there are more than 10 empty price levels
 }
+
+func TestEstimateBoundsOneSided(t *testing.T) {
+	initialMargin := num.DecimalFromFloat(1)
+	riskFactorShort := num.DecimalFromFloat(0.01)
+	riskFactorLong := num.DecimalFromFloat(0.01)
+	linearSlippageFactor := num.DecimalFromFloat(0)
+	sqrter := NewSqrter()
+
+	lowerPrice := num.NewUint(900)
+	basePrice := num.NewUint(1000)
+	upperPrice := num.NewUint(1100)
+	leverageUpper := num.DecimalFromFloat(2.00)
+	leverageLower := num.DecimalFromFloat(2.00)
+	balance := num.NewUint(100)
+
+	// no upper bound supplied
+	metrics := EstimateBounds(
+		sqrter,
+		lowerPrice,
+		basePrice,
+		nil,
+		leverageLower,
+		leverageUpper,
+		balance,
+		linearSlippageFactor,
+		initialMargin,
+		riskFactorShort,
+		riskFactorLong,
+		num.DecimalOne(),
+		num.DecimalOne(),
+		0,
+	)
+	assert.True(t, metrics.LiquidationPriceAtUpper.IsZero())
+	assert.True(t, metrics.PositionSizeAtUpper.IsZero())
+	assert.True(t, metrics.LossOnCommitmentAtUpper.IsZero())
+
+	// no lower bound supplied
+	metrics = EstimateBounds(
+		sqrter,
+		nil,
+		basePrice,
+		upperPrice,
+		leverageLower,
+		leverageUpper,
+		balance,
+		linearSlippageFactor,
+		initialMargin,
+		riskFactorShort,
+		riskFactorLong,
+		num.DecimalOne(),
+		num.DecimalOne(),
+		0,
+	)
+	assert.True(t, metrics.LiquidationPriceAtLower.IsZero())
+	assert.True(t, metrics.PositionSizeAtLower.IsZero())
+	assert.True(t, metrics.LossOnCommitmentAtLower.IsZero())
+
+}

--- a/datanode/service/market_depth_amm.go
+++ b/datanode/service/market_depth_amm.go
@@ -233,8 +233,11 @@ func (m *MarketDepth) getReference(ctx context.Context, marketID string) (num.De
 }
 
 func (m *MarketDepth) expandByLevels(pool entities.AMMPool, levels []*level, priceFactor num.Decimal) ([]*types.Order, []bool, error) {
-	// get positions
+	if len(levels) == 0 {
+		return nil, nil, nil
+	}
 
+	// get position
 	pos, err := m.getAMMPosition(pool.MarketID.String(), pool.AmmPartyID.String())
 	if err != nil {
 		return nil, nil, err

--- a/datanode/service/market_depth_amm_test.go
+++ b/datanode/service/market_depth_amm_test.go
@@ -472,6 +472,34 @@ func TestFairgroundAMM(t *testing.T) {
 	assert.Equal(t, "12956", mds.service.GetBestAskPrice(marketID).String())
 }
 
+func TestAMMDepthOutOfRange(t *testing.T) {
+	ctx := context.Background()
+
+	cfg := service.MarketDepthConfig{
+		AmmFullExpansionPercentage: 200,
+		AmmEstimatedStepPercentage: 0,
+		AmmMaxEstimatedSteps:       0,
+	}
+	mds := getServiceWithConfig(t, cfg)
+	defer mds.ctrl.Finish()
+
+	marketID := vgcrypto.RandomHash()
+
+	ensureLiveOrders(t, mds, marketID)
+	ensureDecimalPlaces(t, mds, 1, 1)
+
+	// set the mid-price to a value no where near the AMM's prices
+	mds.marketData.EXPECT().GetMarketDataByID(gomock.Any(), gomock.Any()).Times(1).Return(entities.MarketData{MidPrice: num.DecimalFromInt64(10)}, nil)
+
+	// data node is starting from network history, initialise market-depth based on whats aleady there
+	ensureAMMs(t, mds, marketID)
+	mds.service.Initialise(ctx)
+
+	assert.Equal(t, 0, int(mds.service.GetTotalAMMVolume(marketID)))
+	assert.Equal(t, 0, int(mds.service.GetAMMVolume(marketID, true)))
+	assert.Equal(t, 0, int(mds.service.GetAMMVolume(marketID, false)))
+}
+
 func ensureLiveOrders(t *testing.T, mds *MDS, marketID string) {
 	t.Helper()
 	mds.orders.EXPECT().GetLiveOrders(gomock.Any()).Return([]entities.Order{

